### DISCUSSION TITLE
Issue in pulling lap time data and inserting it into excel sheet.

### DISCUSSION THREAD

**Discussion by laurence9899** [theOehrly/Fast-F1#442] [Q&A] [answered]
Hi,

I'm trying to pull lap time data for all 20 drivers for all laps and place it into an excel sheet. I have it almost working but am encountering an error when indexing the lap time data. Any suggestions on how to fix this would be greatly appreciated.
Thank you.
My code is below.
#Imports

import fastf1
import openpyxl
import numpy as np

#Setup
path=str('F1RaceDataTable.xlsx')
file_object=openpyxl.load_workbook(path.strip())
sheet_obj=file_object.active

fastf1.Cache.enable_cache('/Users/FYP/FastF1') 

r1=fastf1.get_session(2023, 1,'r')
r1.load(laps=True)
drivers=r1.drivers

k=1
for drv in r1.drivers:
    k=k+1
    laptimes=r1.laps.pick_driver(drv).LapTime
    
    
    n=0
    for column in range(k,len(laps)):
        
       row1=sheet_obj.cell(row=k, column=column)
       row1.value=laptimes[n]
       n=n+1 
    
    
    
file_object.save("BAHRAIN_RaceData.xlsx")

**Comment by formulatimer** [comment]
I'm having the same issue and I have opened the issue #443

**Comment by theOehrly** [comment]
@laurence9899 has this problem resolved itself?

**Reply by laurence9899** [reply]
Yes but I'm still having an issue with this line 
row1.value=laptimes[n]
Is this the wrong way off trying to write the lap times into excel sheet?
I am trying to add them, one by one by looping through each drivers lap time from the race.

Thanks Laurence

**Reply by Casper-Guo** [reply, marked as answer]
`Laps` objects inherits from `pd.Dataframe` so you should be able to use [`to_excel`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_excel.html) on them
